### PR TITLE
feat(prettier): integrate

### PR DESCRIPTION
**Pull request by hekike**
- integrate prettier
- change npm install to yarn install

### CODE REVIEW

File: app/templates/_testIndex.js [hunk 1]
```diff
@@ -3,9 +3,7 @@
 // external modules
 const assert = require('chai').assert;
 
-
-describe('<%= data.name %>', function() {
-
+describe('netflixui-datasource', function() {
```
Comment (DonutEspresso) [inline app/templates/_testIndex.js]: I think this line got lost

**Comment by hekike** [comment]
@DonutEspresso good catch!
I integrated prettier to the generator itself, reverted manual style fixes and run prettier on the project to avoid issues like this.

File: .eslintrc [hunk 2]
```diff
@@ -2,116 +2,242 @@
     // eslint: recommended automatically enables most/all rules from the
     // possible errors section and more:
     // http://eslint.org/docs/rules/#possible-errors
-    "extends": "eslint:recommended",
+    "extends": ["plugin:prettier/recommended"],
```
Comment (DonutEspresso) [inline .eslintrc]: Should we move to eslint.yml?

File: .eslintrc [hunk 3]
```diff
@@ -2,116 +2,242 @@
...
+        "no-path-concat": ["error"],
+        "no-process-exit": ["error"],
 
         // stylistic
-        // turn on the few on that aren't handled by JSCS.
-        "consistent-this": [ 2, "self" ],
-        "no-array-constructor": [ 2 ],
-        "no-nested-ternary": [ 2 ],
-        "no-new-object": [ 2 ],
+        "array-bracket-newline": ["error", "consistent" ],
```
Comment (DonutEspresso) [inline .eslintrc]: Isn't styling handled by prettier now? Are these just duplicating prettier rules?

File: Makefile [hunk 4]
```diff
@@ -12,26 +12,21 @@ TMP		:= $(ROOT)/tmp
 # Tools and binaries
 #
 ESLINT		:= $(NODE_BIN)/eslint
-JSCS		:= $(NODE_BIN)/jscs
+PRETTIER		:= $(NODE_BIN)/prettier
```
Comment (DonutEspresso) [inline Makefile]: This spacing looks weird, can you make sure you're using tabs in Makefile?

File: Makefile [hunk 5]
```diff
@@ -59,18 +54,13 @@ githooks: ## Install git pre-push hooks.
 
 
 .PHONY: lint
-lint: node_modules $(ALL_FILES) ## Run lint checker (eslint).
+lint: $(NODE_MODULES) $(ESLINT) $(ALL_FILES) ## Run lint checker (eslint).
```
Comment (DonutEspresso) [inline Makefile:57]: Nice catch!

File: app/templates/_coverageBadge.js [hunk 6]
```diff
@@ -29,8 +29,14 @@ const out = processLines(originalReadmeStr);
...
-
-function processLines(readmeStr) { /* eslint-disable-line require-jsdoc */
+/**
+ * Process lines
+ * @function processLines
+ * @param {String} readmeStr - readme
+ * @returns {String} readme
+ */
+function processLines(readmeStr) {
+    /* eslint-disable-line require-jsdoc */
```
Comment (DonutEspresso) [inline app/templates/_coverageBadge.js]: You can probably remove this line now 👍

File: app/templates/eslintrc [hunk 7]
```diff
@@ -2,174 +2,177 @@
     // eslint: recommended automatically enables most/all rules from the
     // possible errors section and more:
     // http://eslint.org/docs/rules/#possible-errors
-    "extends": "eslint:recommended",
+    "extends": ["plugin:prettier/recommended"],
```
Comment (DonutEspresso) [inline app/templates/eslintrc]: Same thing, should this be a yml file now? Or js file? Whichever is the future?

File: app/templates/prettierrc [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (DonutEspresso) [inline app/templates/prettierrc]: Is prettier moving to yml files too?

**Review by DonutEspresso — CHANGES_REQUESTED** [review]
Left a few comments, thanks for doing this! 🎉

File: .eslintrc (hunk 2 above)
Comment (hekike) [reply .eslintrc]: both json, js and yaml are supported, I migrated it to .js

File: .eslintrc (hunk 3 above)
Comment (hekike) [reply .eslintrc]: Yes, stylistic rules should be removed

**Pull request merged**